lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -1063,32 +872,39 @@ def alerts(self) -> dict:
                 except KeyError:
                     logger.error("Alert rules were found but no usable labels were present")
 
+            _, errmsg = self.tool.validate_alert_rules(alert_rules)
+            if errmsg:
+                if alerts[identifier]:
+                    del alerts[identifier]
+                relation.data[self._charm.app]["event"] = json.dumps({"errors": errmsg})
```

Comment:
* No
* No
* No

This is a "battle-tested" pattern, and `juju show-application` will also show `valid: true` instead of `valid: false`, so there's no confusion about it.